src/remote.c

```diff
@@ -306,37 +327,41 @@ int git_remote_create(git_remote **out, git_repository *repo, const char *name,
 	return error;
 }
 
+int git_remote_create_ext(git_remote **out, const char *url, const git_remote_create_options *opts)
+{
+	return create_internal(out, url, opts);
+}
+
 int git_remote_create_with_fetchspec(git_remote **out, git_repository *repo, const char *name, const char *url, const char *fetch)
 {
-	git_remote *remote = NULL;
 	int error;
+	git_remote_create_options opts = GIT_REMOTE_CREATE_OPTIONS_INIT;
 
 	if ((error = ensure_remote_name_is_valid(name)) < 0)
 		return error;
 
 	if ((error = ensure_remote_doesnot_exist(repo, name)) < 0)
 		return error;
```

Comment:
These two calls are already done in `create_remote_internal` now. I'd like to avoid this duplication